client-java/instrumentation/src/main/java/org/evomaster/client/java/instrumentation/coverage/methodreplacement/classes/MongoClassReplacement.java

```diff
@@ -0,0 +1,76 @@
+    public static <TResult> Object find(Object mongoCollection, Class<TResult> resultClass) {
+        return handleFind("findResultClass", mongoCollection, Arrays.asList(resultClass), null);
+    }
+
+    @Replacement(replacingStatic = false, type = ReplacementType.TRACKER, id = "findBson", usageFilter = UsageFilter.ANY, category = ReplacementCategory.MONGO, castTo = "com.mongodb.client.FindIterable")
+    public static Object find(Object mongoCollection, @ThirdPartyCast(actualType = "org.bson.conversions.Bson") Object bson) {
+        return handleFind("findBson", mongoCollection, Arrays.asList(bson), bson);
+    }
+
+    @Replacement(replacingStatic = false, type = ReplacementType.TRACKER, id = "findBsonResultClass", usageFilter = UsageFilter.ANY, category = ReplacementCategory.MONGO, castTo = "com.mongodb.client.FindIterable")
+    public static <TResult> Object find(Object mongoCollection, @ThirdPartyCast(actualType = "org.bson.conversions.Bson") Object bson, Class<TResult> resultClass) {
+        return handleFind("findBsonResultClass", mongoCollection, Arrays.asList(bson, resultClass), bson);
+    }
+
+    @Replacement(replacingStatic = false, type = ReplacementType.TRACKER, id = "findClientSessionBson", usageFilter = UsageFilter.ANY, category = ReplacementCategory.MONGO, castTo = "com.mongodb.client.FindIterable")
+    public static Object find(Object mongoCollection, @ThirdPartyCast(actualType = "com.mongodb.client.ClientSession") Object clientSession, @ThirdPartyCast(actualType = "org.bson.conversions.Bson") Object bson) {
+        return handleFind("findClientSessionBson", mongoCollection, Arrays.asList(clientSession, bson), bson);
+    }
+
+    @Replacement(replacingStatic = false, type = ReplacementType.TRACKER, id = "findClientSessionBsonResultClass", usageFilter = UsageFilter.ANY, category = ReplacementCategory.MONGO, castTo = "com.mongodb.client.FindIterable")
+    public static <TResult> Object find(Object mongoCollection, @ThirdPartyCast(actualType = "com.mongodb.client.ClientSession") Object clientSession, @ThirdPartyCast(actualType = "org.bson.conversions.Bson") Object bson, Class<TResult> resultClass) {
+        return handleFind("findClientSessionBsonResultClass", mongoCollection, Arrays.asList(clientSession, bson, resultClass), bson);
+    }
+
+    private static Object handleFind(String id, Object mongoCollection, List<Object> args, Object query) {
+        long start = System.currentTimeMillis();
+        try {
+            Method findMethod = retrieveFindMethod(id, mongoCollection);
+            Object result = findMethod.invoke(mongoCollection, args.toArray());
+            long end = System.currentTimeMillis();
+            handleMongo(mongoCollection, query, true, end - start);
+            return result;
+        } catch (IllegalAccessException | InvocationTargetException e) {
+            throw new RuntimeException(e);
```

Comment:
this is changing the SUT by throwing a different exception. see other replacement methods. eg, can use somthing like:

```
try {
      ...
        } catch (IllegalAccessException e) {
            throw new RuntimeException(e);
        } catch (InvocationTargetException e) {
            throw e.getCause();
        }
```

Reply:
otherwise you might break the behavior of the SUT if it has `catch` clauses for particular exceptions

Reply:
Fixed